### PR TITLE
School library: Unit tests

### PR DESCRIPTION
**Pull request by theodoku**
In this Pull Request: 

- [ ] Add unit tests for all the methods in all the classes:

**Book**
**Person**
**Student**
**Teacher**
**Classroom**
**Rental**
all classes were created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  describe '#of_age?' do
+    context 'when the person is 18 years old or older' do
+      it 'returns true' do
+        person = Person.new(18)
+        expect(person.send(:of_age?)).to be true
+      end
+    end
+
+    context 'when the person is younger than 18 years old' do
+      it 'returns false' do
```
Comment (Davidosky007) [inline spec/person_spec.rb]: * as stated in the [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md#:~:text=Add%20unit%20tests%20for%20all%20the%20methods%20in%20all%20the%20classes) kindly Add unit tests for all the methods in all the classes as some methods were not covered in your tests

**Review by Davidosky007 — COMMENTED** [review]
Hi @theodoku @calebchris000 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Unit tests written using Rspec 👍 
* All tests passed 👍 
* No linter errors 👍 

## Required Changes ♻️
* Kindly write tests for all the methods in your classes as some methods were not covered 🙏 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (theodoku) [reply spec/person_spec.rb]: Code refactored to include all methods in class

File: spec/person_spec.rb [hunk 2]
```diff
@@ -2,19 +2,45 @@
...
 RSpec.describe Person do
-  describe '#of_age?' do
-    context 'when the person is 18 years old or older' do
-      it 'returns true' do
-        person = Person.new(18)
-        expect(person.send(:of_age?)).to be true
+    describe '#can_use_services?' do
+      context 'when the person is of age' do
+        it 'returns true' do
+          person = Person.new(20)
```
Comment (TadesseDev) [inline spec/person_spec.rb]: - [Optional] Good job adding the test for the missing methods in the person class as suggested earlier. But I notice you have not tested some of the classes at all. this includes the `BaseDecorator`  and others. Please try to cover all the methods in all classes.

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**